app/services/scene-collections/scene-collections.ts

```diff
@@ -0,0 +1,865 @@
+        return;
+      }
+    }
+
+    // Fall back to creating a new collection
+    await this.create();
+  }
+
+  /**
+   * Writes data to a collection file
+   * @param id The id of the file
+   * @param data The data to write
+   */
+  private writeDataToCollectionFile(id: string, data: string) {
+    const collectionPath = this.getCollectionFilePath(id);
+    return new Promise((resolve, reject) => {
+      fs.writeFile(collectionPath, data, err => {
+        if (err) {
+          reject(err);
+          return;
+        }
+
+        resolve();
+      });
+    });
+  }
+
+  /**
+   * This deloads all scenes and sources and gets the application
+   * ready to load a new config file.  This should only ever be
+   * performed while the application is already in a "LOADING" state.
+   */
+  private async deloadCurrentApplicationState() {
+    if (!this.initialized) return;
+
+    this.disableAutoSave();
+    await this.save();
+
+    // we should remove inactive scenes first to avoid the switching between scenes
+    this.scenesService.scenes.forEach(scene => {
+      if (scene.id === this.scenesService.activeSceneId) return;
+      scene.remove(true);
+    });
+
+    if (this.scenesService.activeScene) {
+      this.scenesService.activeScene.remove(true);
+    }
+
+    this.sourcesService.sources.forEach(source => {
+      if (source.type !== 'scene') source.remove();
+    });
+
+    this.hotkeysService.unregisterAll();
+  }
+
+  /**
+   * Should be called before any loading operations
+   */
+  private startLoadingOperation() {
+    this.appService.startLoading();
+    this.disableAutoSave();
+  }
+
+  /**
+   * Should be called after any laoding operations
+   */
+  private finishLoadingOperation() {
+    this.appService.finishLoading();
+    this.enableAutoSave();
+  }
+
+  /**
+   * Creates the scenes and sources that come in by default
+   * in an empty scene collection.
+   */
+  private setupEmptyCollection() {
+    this.scenesService.createScene('Scene', { makeActive: true });
+    this.setupDefaultAudio();
+  }
+
+  /**
+   * Creates the default audio sources
+   */
+  private setupDefaultAudio() {
+    this.sourcesService.createSource(
+      'DesktopAudioDevice1',
+      'wasapi_output_capture',
+      {},
+      { channel: E_AUDIO_CHANNELS.OUTPUT_1 }
+    );
+
+    this.sourcesService.createSource(
+      'AuxAudioDevice1',
+      'wasapi_input_capture',
+      {},
+      { channel: E_AUDIO_CHANNELS.INPUT_1 }
+    );
+  }
+
+  /**
+   * Creates and persists new collection from the current application state
+   */
+  private async insertCollection(id: string, name: string) {
+    await this.saveCurrentApplicationStateAs(id);
+    this.ADD_COLLECTION(id, name, (new Date()).toISOString());
+  }
+
+  /**
+   * Deletes on the server and removes from the store
+   */
+  private async removeCollection(id: string) {
+    this.DELETE_COLLECTION(id);
+
+    // TODO: Delete the file (for safety and recoverability we soft delete for now)
```

Comment:
Should  this TODO be a part of this PR?

Reply:
Yeah I'm still undecided as to whether we should hard delete files.  I'll make a final decision and remove the comment.